helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -239,11 +239,14 @@ private void validateMaxPartitionsPerInstance(ClusterEvent event,
           // in this instance
           partitionCount++;
           if (partitionCount > maxPartitionPerInstance) {
+            // Enable maintenance rebalancer for this pipeline
```

Comment:
Isn't it better to enable this maintenance mode in the BestPossible state? Is there any reason that we enable it in this stage?

Reply:
M mode is in best possible. Realized it. Keeping it as is.